### PR TITLE
Add type to JWT decode header

### PR DESCRIPTION
**Pull request by AlyHKafoury**
reference :  https://github.com/krakendio/krakend-ce/issues/711

### CODE REVIEW

File: jws.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ type SignatureConfig struct {
 type SignerConfig struct {
 	Alg                string   `json:"alg"`
 	KeyID              string   `json:"kid"`
+	Type               string   `json:"typ"`
```
Comment (kpacha) [inline jws.go:49]: Why are you adding this new field?

Comment (AlyHKafoury) [reply jws.go:49]: to unify the output across different outputs if you think it is not necessary it is totally ok for me to remove it. it is my first commit from many to come I hope so I am open to discussions for sure

Comment (kpacha) [reply jws.go:49]: This struct contains the configuration for the signer middleware. By adding this new field, we could let the users select the type of token to generate. For that, the `SignerOptions` should be created after checking this new field (and the `typ` header should contain whatever the config says. Something like this:
```
if signerCfg.Type != "" {
opts.ExtraHeaders[jose.HeaderKey("typ")] = signerCfg.Type
}
```

**Review by taik0 — CHANGES_REQUESTED** [review]
Hi @AlyHKafoury can you add the requested changes?

Thank you!

**Comment by AlyHKafoury** [comment]
> Hi @AlyHKafoury can you add the requested changes?
> 
> Thank you!

I have added the required changes, can I ask for your review please ?

**Review by kpacha — APPROVED** [review]

**Comment by AlyHKafoury** [comment]
I am fixing the tests problems

**Comment by AlyHKafoury** [comment]
@kpacha is there a correct method to generate the signed JWT payloads for tests ?

**Comment by kpacha** [comment]
There are some keys at the `fixtures` folder. you can use symmetric or asymmetric alg. you can instantiate a Signer (https://github.com/krakend/krakend-jose/blob/master/gin/jose.go#L24) and use it to sign your test token

**Comment by alombarte** [comment]
Hi @AlyHKafoury, we have selected this feature to release with KrakenD 2.5, which will be out really soon. Will you be able to complete the tests?

**Comment by AlyHKafoury** [comment]
Hello,

Yes I will push it in a few hours

Best Regards,
Aly Kafoury

**Comment by alombarte** [comment]
Hello @AlyHKafoury,

We will be preparing the release of KrakenD CE 2.5 early next week. I just wanted to ping you to see if you can complete this. Otherwise we will introduce this functionality in krakenD 2.6 (2024)

**Comment by alombarte** [comment]
Hi @AlyHKafoury,

Are you still interested in finishing this for the next release?

**Comment by AlyHKafoury** [comment]
Yes I will do my best to fix the unit tests I am just not very familiar with the key generation

**Comment by AlyHKafoury** [comment]
@alombarte I made changes to the test case encryption once it is reviewed I believe the code should pass the test case passes locally now

**Comment by AlyHKafoury** [comment]
@kpacha can you please re-review

**Comment by kpacha** [comment]
@AlyHKafoury I've created a PR on your fork fixing the failling tests. Once you approve and merge it, we'll be able to merge and close this PR

**Review by taik0 — APPROVED** [review]

**Comment by kpacha** [comment]
Thanks for the contribution!!!

**Pull request merged**